packages/react/menu/src/Menu.tsx

```diff
@@ -758,13 +765,13 @@ const MenuCheckboxItem = React.forwardRef<MenuCheckboxItemElement, MenuCheckboxI
       <ItemIndicatorProvider scope={props.__scopeMenu} checked={checked}>
         <MenuItem
           role="menuitemcheckbox"
-          aria-checked={checked}
+          aria-checked={isIndeterminate(checked) ? 'mixed' : checked}
           {...checkboxItemProps}
           ref={forwardedRef}
           data-state={getCheckedState(checked)}
           onSelect={composeEventHandlers(
             checkboxItemProps.onSelect,
-            () => onCheckedChange?.(!checked),
+            () => onCheckedChange?.(isIndeterminate(checked) || !checked),
```

Comment:
nit, let's use this form to stay consistent with `Checkbox.tsx`:

```suggestion
            () => onCheckedChange?.(isIndeterminate(checked) ? true : !checked),
```